core/src/main/scala/org/apache/spark/sql/crossdata/XDContext.scala

```diff
@@ -43,23 +43,26 @@ import org.apache.spark.{Logging, SparkContext}
  * and adds some features of the Crossdata system.
  * @param sc A [[SparkContext]].
  */
-class XDContext(@transient val sc: SparkContext) extends SQLContext(sc) with Logging with CoreConfig {
-
+class XDContext(@transient val sc: SparkContext,
```

Comment:
XDContext is one of the main interfaces with the user, so we could provide a more confortable constructor:

class XDContext private(@transient val sc: SparkContext,
                 userConfig: Option[Config]) extends SQLContext(sc) with Logging with CoreConfig {
  self =>

  def this(sc: SparkContext) =
    this(sc, None)

  def this(sc: SparkContext, config: Config) =
    this(sc, Some(config))